src/Mafia/Shell.hs

```diff
@@ -0,0 +1,373 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Mafia.Shell (
+    ShellError(..)
+  , renderShellError
+  , runShell
+  ) where
+
+import           Control.Monad.IO.Class (MonadIO(..))
+
+import qualified Data.Attoparsec.Text as Atto
+import qualified Data.Map as Map
+import qualified Data.Text as T
+
+import           Mafia.Hash
+import           Mafia.Home
+import           Mafia.IO
+import           Mafia.Path
+import           Mafia.Process
+
+import           P
+
+import           System.IO (IO)
+
+import           X.Control.Monad.Trans.Either (EitherT, hoistMaybe, hoistEither)
+
+
+data Script =
+  Script {
+      scriptId :: !Text
+    , scriptHash :: !Hash
+    , scriptPath :: !File
+    , scriptPackages :: ![Package]
+    , scriptSubmodules :: ![Submodule]
+    , scriptMain :: !Text
+    } deriving (Eq, Ord, Show)
+
+newtype Package =
+  Package {
+      unPackage :: Text
```

Comment:
This actually does support constraints, I'm just not parsing them, the string you put in the `PACKAGE` pragma is transplanted directly in to a `.cabal` file. I've updated the cli test to demonstrate.

I don't think they are safe to use in prod anyway because they'd depend on hackage and github being up.